lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1634,6 +1634,18 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
   };
 
   switch (II->getIntrinsicID()) {
+  case Intrinsic::assume: {
+    // llvm.assume translation is currently supported only within
+    // SPV_INTEL_optimization_hints extension, ignore it otherwise, since it's
+    // an optimization hint
+    if (BM->isAllowedToUseExtension(
+            ExtensionID::SPV_INTEL_optimization_hints)) {
+      SPIRVValue *SV = transValue(II->getOperand(0), BB);
+      SV->addDecorate(DecorationAssumeTrueINTEL);
```

Comment:
This seems to be adding the decoration to the _input_ of the intrinsic, which does not seem correct with respect to control flow.

Reply:
Remade it to be an instruction.